scheduler/executor/interfaces/snapshoter.go

```diff
@@ -0,0 +1,21 @@
+package interfaces
+
+import (
+	"time"
+)
+
+// Snapshoter used by the scheduler for creating, updating and reading snapshots of the schedules.
+type Snapshoter interface {
```

Comment:
```suggestion
type Executor interface {
```
The folder `snapshot` you dont have to put everything in `interfaces`